code/game/turfs/closed/walls.dm

```diff
@@ -34,48 +34,63 @@
 
 	var/list/dent_decals
 
-/turf/closed/wall/MouseDrop_T(mob/living/carbon/carbon_mob, mob/user)
+/turf/closed/wall/MouseDrop_T(atom/dropping, mob/user, params)
 	..()
-	if(carbon_mob != user)
+	if(dropping != user)
 		return
-	if(carbon_mob.is_leaning == TRUE)
+	if(!iscarbon(dropping) && !iscyborg(dropping))
 		return
-	if(carbon_mob.pulledby)
+	var/mob/living/leaner = dropping
+	if(leaner.incapacitated(IGNORE_RESTRAINTS) || leaner.stat != CONSCIOUS || HAS_TRAIT(leaner, TRAIT_NO_TRANSFORM))
```

Comment:
we can move the `HAS_TRAIT()` check well before the typecast since traits work on the datum level. granted it might not make sense given the preceding code but it is definitely possible.

Reply:
Same as following